posthog/queries/retention/retention_events_query.py

```diff
@@ -151,7 +163,7 @@ def get_query(self) -> Tuple[str, Dict[str, Any]]:
         query = f"""
             SELECT {','.join(_fields)} FROM events {self.EVENT_TABLE_ALIAS}
             {sample_clause}
-            {self._get_distinct_id_query()}
+            {self._get_person_id_overrides_query() if self._using_person_on_events_v2 else self._get_distinct_id_query()}
             {person_query}
```

Comment:
This person_query seems to be controlled by https://github.com/PostHog/posthog/blob/master/posthog/queries/retention/retention_events_query.py#L195 - only v1 person on events.

Is the idea to always enable v1 before v2 and v2 takes priority if enabled?

Reply:
v2 will take priority over everything if enabled yes. And v1 and v2 share the property that person property lookups happen the same way, only person ID changes

Reply:
I guess without seeing how it's called, this is a bit confusing. Easy to make mistakes here (i.e. remember to pass in `v1 or v2` to `_using_person_on_events` everywhere)

Not asking for any action, just making this explicit to prevent bugs down the line / help you figure out a more bullet-proof way

Reply:
So currently we have `using_person_on_events` tagged everywhere. With the approach I've followed here we'd have to add `using_person_on_events_v2` everywhere the other one is. Do you think being explicit like that is better or should I refactor everything to receive some sort of `mode` that is either `standard`, `poev1`, or `poev2`?